test/hotspot/jtreg/compiler/lib/ir_framework/test/TestVM.java

```diff
@@ -883,9 +885,15 @@ private void runTests() {
 
         // Print execution times
         if (VERBOSE || PRINT_TIMES) {
-            System.out.println(System.lineSeparator() + System.lineSeparator() + "Test execution times:");
+            // Here we are in the test VM, which was spawned by the main jtreg VM. the stdout of the test VM is hidden
+            // unless the Verbose or ReportStdout flag are used. TestFrameworkSocket is used by the parent jtreg
+            // VM and the test VM to communicate. By sending the prints through the TestFrameworkSocket with the
+            // parameter stdout set to true, the parent VM will print the received messages to its stdout, making it
+            // visible to the user.
```

Comment:
Maybe you can move this comment directly to `TestFrameworkSocket.write()` as documentation for the other code as well that uses the socket to dump messages.